src/main/java/org/apache/sysds/runtime/instructions/fed/CovarianceFEDInstruction.java

```diff
@@ -0,0 +1,253 @@
+import org.apache.sysds.runtime.controlprogram.federated.FederationUtils;
+import org.apache.sysds.runtime.functionobjects.COV;
+import org.apache.sysds.runtime.instructions.InstructionUtils;
+import org.apache.sysds.runtime.instructions.cp.CM_COV_Object;
+import org.apache.sysds.runtime.instructions.cp.CPOperand;
+import org.apache.sysds.runtime.instructions.cp.Data;
+import org.apache.sysds.runtime.instructions.cp.DoubleObject;
+import org.apache.sysds.runtime.instructions.cp.ScalarObject;
+import org.apache.sysds.runtime.matrix.data.MatrixBlock;
+import org.apache.sysds.runtime.matrix.operators.COVOperator;
+import org.apache.sysds.runtime.matrix.operators.Operator;
+
+public class CovarianceFEDInstruction extends BinaryFEDInstruction {
+	private CovarianceFEDInstruction(Operator op, CPOperand in1, CPOperand in2, CPOperand out, String opcode,
+		String istr) {
+		super(FEDInstruction.FEDType.AggregateBinary, op, in1, in2, out, opcode, istr);
+	}
+
+	private CovarianceFEDInstruction(Operator op, CPOperand in1, CPOperand in2, CPOperand in3, CPOperand out,
+		String opcode, String istr) {
+		super(FEDInstruction.FEDType.AggregateBinary, op, in1, in2, in3, out, opcode, istr);
+	}
+
+
+	public static CovarianceFEDInstruction parseInstruction(String str) {
+		CPOperand in1 = new CPOperand("", Types.ValueType.UNKNOWN, Types.DataType.UNKNOWN);
+		CPOperand in2 = new CPOperand("", Types.ValueType.UNKNOWN, Types.DataType.UNKNOWN);
+		CPOperand in3 = null;
+		CPOperand out = new CPOperand("", Types.ValueType.UNKNOWN, Types.DataType.UNKNOWN);
+
+		String[] parts = InstructionUtils.getInstructionPartsWithValueType(str);
+		String opcode = parts[0];
+
+		if( !opcode.equalsIgnoreCase("cov") ) {
+			throw new DMLRuntimeException("CovarianceCPInstruction.parseInstruction():: Unknown opcode " + opcode);
+		}
+
+		COVOperator cov = new COVOperator(COV.getCOMFnObject());
+		if ( parts.length == 4 ) {
+			parseBinaryInstruction(str, in1, in2, out);
+			return new CovarianceFEDInstruction(cov, in1, in2, out, opcode, str);
+		} else if ( parts.length == 5 ) {
+			in3 = new CPOperand("", Types.ValueType.UNKNOWN, Types.DataType.UNKNOWN);
+			parseBinaryInstruction(str, in1, in2, in3, out);
+			return new CovarianceFEDInstruction(cov, in1, in2, in3, out, opcode, str);
+		}
+		else {
+			throw new DMLRuntimeException("Invalid number of arguments in Instruction: " + str);
+		}
+	}
+
+	@Override
+	public void processInstruction(ExecutionContext ec) {
+		MatrixObject mo1 = ec.getMatrixObject(input1);
+		MatrixObject mo2 = ec.getMatrixObject(input2);
+		COVOperator cop = ((COVOperator)_optr);
+
+		if(mo1.isFederated() && mo2.isFederated() && !mo1.getFedMapping().isAligned(mo2.getFedMapping(), false))
+			throw new DMLRuntimeException("Not supported matrix-matrix binary operation: covariance.");
+
+		FederatedRequest fr2 = null;
+		if(mo1.isFederated() && mo2.isFederated() && mo1.getFedMapping().isAligned(mo2.getFedMapping(), false)) {
```

Comment:
This if else can also be put into individual methods.
because just looking at this, i have to figure out what is inside each if statements, while if you have methods it would be more explanatory.